### DISCUSSION TITLE
cache is not working

### DISCUSSION THREAD

**Discussion by jason-liu22** [vercel/swr#2458] [Q&A]
The infinite loading example in [SWR vercel app](https://swr.vercel.app/examples/infinite-loading) is wonderful.
I tried it with my api but it looks like that cache is not working.
I am using it in my nextjs application.
But data is re-fetched every time (every focus, and key value change) in all the component used with swr.
Here is the example.
https://codesandbox.io/s/swr-infinite-forked-50ohd0

**Comment by koba04** [comment]
@endo-aki22 
Yes, that's expected behavior and you can disable it with options like `revalidateOnFocus`, `revalidateIfStale`, and etc.
https://swr.vercel.app/docs/api#options

`useSWRInfinte` also revalidates its first page and you can disable it with the `revalidateFirstPage` option.

https://swr.vercel.app/docs/pagination#parameters

**Reply by jason-liu22** [reply]
Thanks. 
I can disable those options. but cache is not working.
The example in [SWR vercel app](https://swr.vercel.app/examples/infinite-loading) doesn't work with cache in my case.
https://codesandbox.io/s/swr-infinite-forked-50ohd0

**Reply by koba04** [reply]
Could you explain your expected behavior?

> but cache is not working.
> The example in [SWR vercel app](https://swr.vercel.app/examples/infinite-loading) doesn't work with cache in my case.

**Reply by jason-liu22** [reply]
![02-27-23-00-34-35](https://user-images.githubusercontent.com/61251512/221501936-fcfd2a3e-a57a-47f3-bb70-1b190e9e0cf8.png)
![02-27-23-00-35-08](https://user-images.githubusercontent.com/61251512/221502031-657921e4-eaab-452f-a2b0-100424b74553.png)
The second one is using my API.
https://codesandbox.io/s/swr-infinite-forked-50ohd0

**Reply by jason-liu22** [reply]
As you can see in the network tab, in my app case, the cache is not working.

**Reply by koba04** [reply]
SWR fetches data while having stale data, and you can disable the behavior with options.

```js
    {
      revalidateOnFocus: false,
      revalidateIfStale: false,
      revalidateFirstPage: false
    }
```

https://codesandbox.io/s/swr-infinite-forked-rtyou0?file=/src/App.js

**Reply by jason-liu22** [reply]
but the swr vercel app is working well as expected.
I just replaced it with my API.

**Reply by jason-liu22** [reply]
https://swr.vercel.app/docs/pagination#parameters
[My app](https://codesandbox.io/s/swr-infinite-forked-rtyou0?file=/src/App.js) is using same code from the swr vercel app above. I just use my backend API.

**Reply by jason-liu22** [reply]
The swr vercel app is working well with cache without setting the those options as you can see the image uploaded above.

**Reply by koba04** [reply]
The last request in the second diagram seems to be caused by re-focusing on the page (by clicking the window etc).

**Reply by jason-liu22** [reply]
yes. whatever, the sample is working well with cache. but not in my app.

**Reply by koba04** [reply]
This also happens in the example app because it's the expected behavior. You can see it after re-focusing on the page after 5 seconds.
https://swr.vercel.app/examples/infinite-loading

**Reply by jason-liu22** [reply]
No. you didn't get what I mean. It's not about 5 seconds. It even happens every seconds.
you can check the sample and my app in the network tab.
The sample use the cache every focus. but my app re-fetch data every focus. not use cache.
